spec/tagManagerWrap.spec.js

```diff
@@ -0,0 +1,184 @@
+import TagManager from '../src/tagmanager-wrap';
+
+describe('Tagmanager', () => {
+  const tagManager = new TagManager([], {
+    gtmId: '222222',
+    startPush: {
+      page_type: 'pages:demo',
+      path_category: 'reformas-e-reparos/demo',
+      experiments: window.experiments || [],
+    },
+  });
+
+  describe('constructor', () => {
+    it('defines default properties', () => {
+      expect(tagManager.options).toBeDefined();
+      expect(tagManager.dataLayer).toBeDefined();
+    });
+  });
+
+  describe('.defineProperties', () => {
+    it('sets default properties', () => {
+      expect(tagManager.options.gtmId).toEqual('222222');
+      expect(tagManager.options.startPush.page_type).toEqual('pages:demo');
+      expect(tagManager.options.startPush.path_category).toEqual('reformas-e-reparos/demo');
+      expect(tagManager.options.virtualPageViewEvent).toEqual('virtual_pageview');
+    });
+  });
+
+  describe('.dataLayer init', () => {
+    it('sets initial data', () => {
+      expect(tagManager.dataLayer.length).not.toEqual(0);
+    });
+  });
+
+  describe('.init', () => {
+    beforeEach(() => {
+      document.body.innerHTML = __html__['spec/fixtures/index.html'];
+      tagManager.init();
+    });
+
+    it('appends async script', () => {
+      const scriptTag = document.querySelector('[data-id="google-tagmanager"]');
+
+      expect(scriptTag).not.toBeNull();
+    });
+
+    it('appends noscript fallback', () => {
```

Comment:
@eduardojmatos @felipeorlando @williamcosta @nathpaiva Testing if actual noscript was added to the page